pylint_nautobot/q_search_filter.py

```diff
@@ -0,0 +1,48 @@
+"""Check for use of SearchFilter on NautobotFilterSets instead of custom `q` search functions."""
+
+from astroid import Assign, AssignName, ClassDef
+from pylint.checkers import BaseChecker
+
+from .utils import is_version_compatible
+
+_META_CLASSES = {
+    "nautobot.extras.filters.NautobotFilterSet": ">1",
+    "nautobot.apps.filters.NautobotFilterSet": ">1",
+}
+
+
+class NautobotUseSearchFilterChecker(BaseChecker):
+    """Visit NautobotFilterSet subclasses and check for use of `q = SearchFilter`, instead of `q = django_filters.CharField`."""
+
+    version_specifier = ">=2,<3"
+
+    name = "nautobot-use-search-filter"
+    msgs = {
+        "C4272": (
+            "Use `q = SearchFilter` instead of django_filters.CharField and the custom `search` function.",
+            "nb-use-search-filter",
+            "Nautobot provides a `SearchFilter` class that uses MappedPredicates to provide a more flexible search experience. "
+            "This should be used in place of `django_filters.CharFilter` and no longer requires the custom `search` function.",
```

Comment:
Should we *also* have a checker for `def search(...)` as well?

Reply:
We should, and I discovered after this pr that you can combine multiple checks, so I will add this to it.

Reply:
Implemented a checker for `def search(...)`.